### PR TITLE
icq relayer integration into local-interchain

### LINKED ISSUE #12: set up ICQ relayer on local-ic

### PR DESCRIPTION
**Pull request by bekauz**
closes #12 
branched off from #15 , so review this after that is merged

summary:
1. spin up standard local-ic docker network
2. generate an `.env` file needed to pass to the ICQ relayer docker img on startup, write it to a local file
3. `docker inspect` the neutron container, grab its parent docker network id and its mountpoint on the host
4. `docker run` the icq relayer, passing the previously generated `.env` file, attaching it to the same docker network on the host machine as neutron, and mounting it to the same volume as the neutron (to share the keyring)

as it's the first iteration of integrating ICQ into our testing suite, it's heavily inspired by neutron ICQ docs and the [ICQ relayer repo](https://github.com/neutron-org/neutron-query-relayer). it will be refined with future prs that will adapt this initial setup to be used by orbital.

### CODE REVIEW

File: contracts/orbital-core/Cargo.toml [hunk 1]
```diff
@@ -29,3 +29,4 @@ protobuf        = { workspace = true }
 prost           = { workspace = true }
 prost-types     = { workspace = true }
 serde-json-wasm = { workspace = true }
+cosmos-sdk-proto= { workspace = true }
```
Comment (Art3miX) [inline contracts/orbital-core/Cargo.toml:32]: indentation

File: contracts/orbital-core/src/icq.rs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    query_id: u64,
+    _height: Height,
+    data: Binary,
+) -> StdResult<Response<NeutronMsg>> {
+    // Decode the transaction data
+    let tx: TxRaw = TxRaw::decode(data.as_slice())
+        .map_err(|_| StdError::generic_err("sudo_tx_query_result failed to decode tx_raw"))?;
+    let body: TxBody = TxBody::decode(tx.body_bytes.as_slice())
+        .map_err(|_| StdError::generic_err("sudo_tx_query_result failed to decode tx_body"))?;
+
```
Comment (Art3miX) [inline contracts/orbital-core/src/icq.rs:82]: Why neutron didn't provide helpers or those?

File: contracts/orbital-core/src/icq.rs [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+            StdError::generic_err("sudo_tx_query_result failed to get registered query response")
+        })?;
+    let transactions_filter = registered_query.registered_query.transactions_filter;
+
+    #[allow(clippy::match_single_binding)]
+    // Depending of the query type, check the transaction data to see whether is satisfies
+    // the original query. If you don't write specific checks for a transaction query type,
+    // all submitted results will be treated as valid.
+    //
+    // TODO: come up with solution to determine transactions filter type
```
Comment (Art3miX) [inline contracts/orbital-core/src/icq.rs:95]: probably try parse into struct / enum ?

File: contracts/orbital-core/src/icq.rs [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    deps.api.debug(
+        format!(
+            "WASMDEBUG: sudo_kv_query_result received; query_id: {:?}",
+            query_id,
+        )
+        .as_str(),
+    );
+
+    // TODO: provide an actual example. Currently to many things are going to change
+    // after @pro0n00gler's PRs to implement this.
```
Comment (Art3miX) [inline contracts/orbital-core/src/icq.rs:212]: Would hope there will be a helper package that will hide a lot of those stuff.

File: contracts/orbital-core/src/msg.rs [hunk 5]
```diff
@@ -36,6 +44,9 @@ pub enum QueryMsg {
 
     #[returns(Option<String>)]
     ClearingAccountAddress { addr: String, domain: String },
+
+    #[returns(neutron_sdk::interchain_queries::v047::queries::BalanceResponse)]
+    Balance { query_id: u64 },
```
Comment (Art3miX) [inline contracts/orbital-core/src/msg.rs:49]: I think returning our own type here will be better.

File: local-interchaintest/src/main.rs [hunk 6]
```diff
@@ -41,18 +43,25 @@ fn main() -> Result<(), Box<dyn Error>> {
...
+    // start the icq relayer. this runs in detached mode so we need
+    // to manually kill it before each run for now.
+    start_icq_relayer()?;
 
+    let mut uploader = test_ctx.build_tx_upload_contracts();
     let orbital_core_local_path = format!("{}/artifacts/orbital_core.wasm", current_dir.display());
 
+    info!("sleeping to allow icq relayer to start...");
+    std::thread::sleep(Duration::from_secs(30));
+
```
Comment (Art3miX) [inline local-interchaintest/src/main.rs:64]: Really hate seeing sleep lol, isn't there a way to know if the relayer is up?

File: local-interchaintest/src/utils.rs [hunk 7]
```diff
@@ -67,6 +69,29 @@ pub fn query_user_config(
     Ok(user_config)
 }
 
+pub fn query_balance_query_id(
```
Comment (Art3miX) [inline local-interchaintest/src/utils.rs:72]: weird name lol

File: local-interchaintest/src/utils.rs [hunk 8]
```diff
@@ -138,3 +195,128 @@ pub fn admin_register_domain(
...
+    );
+    let env_content = format!(
+        r#"
+RELAYER_NEUTRON_CHAIN_RPC_ADDR={neutron_rpc}
+RELAYER_NEUTRON_CHAIN_REST_ADDR={neutron_rest}
+RELAYER_NEUTRON_CHAIN_HOME_DIR=/data
+RELAYER_NEUTRON_CHAIN_SIGN_KEY_NAME=acc3
+RELAYER_NEUTRON_CHAIN_GAS_PRICES=0.5untrn
+RELAYER_NEUTRON_CHAIN_GAS_LIMIT=10000000
+RELAYER_NEUTRON_CHAIN_GAS_ADJUSTMENT=2.0
```
Comment (Art3miX) [inline local-interchaintest/src/utils.rs:223]: 😱

**Review by Art3miX — APPROVED** [review]

**Review by keyleu — APPROVED** [review]

File: local-interchaintest/src/main.rs (hunk 6 above)
Comment (bekauz) [reply local-interchaintest/src/main.rs:64]: hmm, maybe we could query the logs of that docker container and sleep until a certain log comes up? but in general it's a separate docker image and it runs in detached mode so nothing really gets returned from there..

**Pull request merged**